sklearn/preprocessing/_encoders.py

```diff
@@ -651,18 +772,31 @@ def inverse_transform(self, X):
         found_unknown = {}
 
         for i in range(n_features):
-            n_categories = len(self.categories_[i])
+            if self.drop is None or self.drop_idx_[i] is None:
+                cats = self.categories_[i]
+            else:
+                cats = np.delete(self.categories_[i], self.drop_idx_[i])
+            n_categories = len(cats)
+            if n_categories == 0:
+                X_tr[:, i] = self.categories_[i][self.drop_idx_[i]]
+                j += n_categories
+                continue
             sub = X[:, j:j + n_categories]
-
             # for sparse X argmax returns 2D matrix, ensure 1D array
             labels = np.asarray(_argmax(sub, axis=1)).flatten()
-            X_tr[:, i] = self.categories_[i][labels]
-
-            if self.handle_unknown == 'ignore':
-                # ignored unknown categories: we have a row of all zero's
+            X_tr[:, i] = cats[labels]
+            if self.handle_unknown == 'ignore' or self.drop is not None:
```

Comment:
Please consider keeping these 2 cases separate, like they are in https://github.com/scikit-learn/scikit-learn/pull/12884

These 2 cases are supposed to be mutually exclusive, and the only portion of code they have in common is the computation of 
`                unknown = np.asarray(sub.sum(axis=1) == 0).flatten()`

which is not a reason for a refactoring IMO. It's unclear at first sight why they are treated at the same place. Moreover `unknown` is not an appropriate name here if we're in the `drop is not None` case.